docs/Measuring_Resonances.md

```diff
@@ -65,45 +41,116 @@ the system that may damage the electronics.
+too little RAM, the installation may fail, and you will need to
+[enable swap](https://nebl.io/neblio-university/enabling-increasing-raspberry-pi-swap/).
 
-Next, run the following commands to install the additional dependencies:
-```
-sudo apt update
-sudo apt install python-numpy python-matplotlib
-```
+## Wiring Option 1: Direct Raspberry Pi Connection
+
+The recommended way to connect the sensor to the printer is to connect it directly to
+the SPI bus on the Raspberry Pi.
+
+You need to connect ADXL345 to your Raspberry Pi via SPI. Note that the I2C
+connection, which is suggested by ADXL345 documentation, has too low throughput
+and **will not work**. The recommended connection scheme:
+
+| ADXL345 pin  | RPi pin | RPi pin name |
+| :--:         | :--:    | :--: |
+| 3V3 (or VCC) | 01      | 3.3v DC power |
+| GND          | 06      | Ground |
+| CS           | 24      | GPIO08 (SPI0_CE0_N) |
+| SDO          | 21      | GPIO09 (SPI0_MISO) |
+| SDA          | 19      | GPIO10 (SPI0_MOSI) |
+| SCL          | 23      | GPIO11 (SPI0_SCLK) |
+
+Fritzing wiring diagrams for some of the ADXL345 boards:
+
+![ADXL345-Rpi](img/adxl345-fritzing.png)
 
-Afterwards, check and follow the instructions in the
-[RPi Microcontroller document](RPi_microcontroller.md) to setup the
-"linux mcu" on the Raspberry Pi.
+Double-check your wiring before powering up the Raspberry Pi to prevent
+damaging it or the accelerometer.
 
-Make sure the Linux SPI driver is enabled by running `sudo
-raspi-config` and enabling SPI under the "Interfacing options" menu.
+Check and follow the
+instructions in the [RPi Microcontroller document](RPi_microcontroller.md) to setup the
+"linux mcu" on the Raspberry Pi. Make sure the Linux SPI driver is enabled by running
+`sudo raspi-config` and enabling SPI under the "Interfacing options" menu.
 
 Add the following to the printer.cfg file:
+
 ```
 [mcu rpi]
 serial: /tmp/klipper_host_mcu
 
 [adxl345]
-cs_pin: rpi:None
+cs_pin: rpi: None
 
 [resonance_tester]
 accel_chip: adxl345
 probe_points:
-    100,100,20  # an example
+    100,100,20  # It is advised to start with 1 probe point, in the middle
+    of the print bed, slightly above it.
 ```
-It is advised to start with 1 probe point, in the middle of the print bed,
-slightly above it.
+
+Restart Klipper via the `RESTART` command.
+
+## Wiring Option 2: MCU Connection (Advanced Users)
+If you want to avoid installing the Linux MCU software on the Raspberry Pi, your Pi's
+GPIO is already in use, or some other reason prevents you from using option 1, you can
+connect the accelerometer to your Primary MCU, or even a secondary MCU. There are some
+caveats, however:
+* As most printer boards output 5V power for accessories, you'll likely need an ADXL345
+board with a built in voltage level shifter.
+* The ADXL345 does not adhere to the SPI spec and cannot share an SPI bus with another
+device. (Thanks @dmbutyugin for pointing that out!) The sensor will need to be the only
```

Comment:
No need to give credit to me in the docs :) Also, it was actually Kevin who first pointed out this defect of ADXL345.